pkg/controllers/cleanup/controller.go

```diff
@@ -0,0 +1,76 @@
+package cleanup
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	kyvernov1alpha1informers "github.com/kyverno/kyverno/pkg/client/informers/externalversions/kyverno/v1alpha1"
+	kyvernov1alpha1listers "github.com/kyverno/kyverno/pkg/client/listers/kyverno/v1alpha1"
+	"github.com/kyverno/kyverno/pkg/clients/dclient"
+	controllerutils "github.com/kyverno/kyverno/pkg/utils/controller"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+type Controller struct {
+	client dclient.Interface
+	// listers
+	cpolLister kyvernov1alpha1listers.ClusterCleanupPolicyLister
+	polLister  kyvernov1alpha1listers.CleanupPolicyLister
+
+	// queue
+	queue workqueue.RateLimitingInterface
+}
+
+const (
+	MaxRetries = 10
+	Workers    = 3
+)
+
+func NewController(dclient dclient.Interface, cpolInformer kyvernov1alpha1informers.ClusterCleanupPolicyInformer, polInformer kyvernov1alpha1informers.CleanupPolicyInformer) *Controller {
+	c := &Controller{
+		client:     dclient,
+		cpolLister: cpolInformer.Lister(),
+		polLister:  polInformer.Lister(),
+		queue:      workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "cleanup-controller"),
+	}
+
+	cpolInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: c.enqueue,
+		UpdateFunc: func(_, obj interface{}) {
+			c.enqueue(obj)
+		},
+	})
+	polInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: c.enqueue,
+		UpdateFunc: func(_, obj interface{}) {
+			c.enqueue(obj)
+		},
+	})
+	return c
+}
+
+func (c *Controller) enqueue(obj interface{}) {
+	c.queue.Add(obj)
+}
+
+func (c *Controller) Run(ctx context.Context, workers int) {
+	controllerutils.Run(ctx, logger.V(3), ControllerName, time.Second, c.queue, Workers, MaxRetries, c.reconcile)
+}
+
+func (c *Controller) reconcile(ctx context.Context, logger logr.Logger, key, namespace, name string) error {
+	policy, err := c.getPolicy(namespace, name)
+	if err != nil {
+		logger.Error(err, "unable to get the policy from policy informer")
+		return err
+	}
+
+	cronjob := getCronJobForTriggerResource(policy)
+	_, err = c.client.CreateResource("batch/v1", "CronJob", "default", cronjob, false)
```

Comment:
We should use a kube client here, not a dynamic one.

Reply:
I am using Kubeclient now, but still, CronJob is not been created.

Reply:
@NikhilSharmaWe see my other comment